File: app/src/main/java/com/montunosoftware/pillpopper/android/view/CalendarViewPager.java
package com.montunosoftware.pillpopper.android.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.core.view.ViewCompat;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

/**
 * @author
 * Created by adhithyaravipati on 4/11/16.
 */
public class CalendarViewPager extends ViewPager {

    public CalendarViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {

        int heightMeasure;
        int height = 0;
        for(int i = 0; i < getChildCount(); i++) {
            View child = getChildAt(i);
            child.measure(widthMeasureSpec, MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED));
            int h = child.getMeasuredHeight();
            if(h > height) height = h;
        }

        heightMeasure = MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY);

        super.onMeasure(widthMeasureSpec, heightMeasure);
    }

    @Override
    public void setAdapter(PagerAdapter adapter) {
        super.setAdapter(adapter);
        if(getResources().getConfiguration().getLayoutDirection() == View.LAYOUT_DIRECTION_RTL){
            setCurrentItem(adapter.getCount()-1);
        }
    }

    @Override
    public void setLayoutDirection(int layoutDirection) {
        if(layoutDirection == ViewCompat.LAYOUT_DIRECTION_LTR) {
            setRotationY(0);
        }
        else if(layoutDirection == ViewCompat.LAYOUT_DIRECTION_RTL) {
            setRotationY(180);
        }
    }
}

File: app/src/main/java/com/montunosoftware/pillpopper/kotlin/calendarviewpager/CalendarViewPager.kt
package com.montunosoftware.pillpopper.kotlin.calendarviewpager

import android.content.Context
import android.util.AttributeSet
import androidx.core.view.ViewCompat
import androidx.viewpager.widget.ViewPager

class HistoryCalendarViewPager(context: Context, attrs: AttributeSet?) : ViewPager(context, attrs) {

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val heightMeasure: Int
        var height = 0
        for (i in 0 until childCount) {
            val child = getChildAt(i)
            child.measure(widthMeasureSpec, MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED))
            val h = child.measuredHeight
            if (h > height) height = h
        }
        heightMeasure = MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY)
        super.onMeasure(widthMeasureSpec, heightMeasure)
    }

    override fun setLayoutDirection(layoutDirection: Int) {
        if (layoutDirection == ViewCompat.LAYOUT_DIRECTION_LTR) {
            rotationY = 0f
        } else if (layoutDirection == ViewCompat.LAYOUT_DIRECTION_RTL) {
            rotationY = 180f
        }
    }
}